app/src/main/java/com/sun/premierleague/ui/team/TeamPresenter.kt

```diff
@@ -0,0 +1,62 @@
+package com.sun.premierleague.ui.team
+
+import com.sun.premierleague.R
+import com.sun.premierleague.data.model.Coache
+import com.sun.premierleague.data.model.Player
+import com.sun.premierleague.data.model.TeamItem
+import com.sun.premierleague.data.source.FootballRepository
+import com.sun.premierleague.data.source.remote.OnDataLoadedCallback
+import java.lang.Exception
+
+class TeamPresenter(
+    private val view: TeamContact.View,
+    private val repository: FootballRepository
+) : TeamContact.Presenter {
+
+    override fun getTeamInformation(idTeam: String) {
+        repository.getTeam(idTeam, object : OnDataLoadedCallback<List<TeamItem>> {
+            override fun onSuccess(data: List<TeamItem>) {
+                if (data.isEmpty()) {
+                    view.showError(R.string.err_no_data)
+                    return
+                }
+                view.showTeam(data)
+
+                val coaches: List<Coache> = data.first().coaches
+                view.showCoach(coaches)
+
+                val players: List<Player> = data.first().players
+                if (players.isEmpty()) {
+                    view.showError(R.string.err_no_player)
+                    return
+                }
+                val defenders = filterData(players, DEFENDERS)
+                val forwarders = filterData(players, FORWARDS)
+                val goalkeepers = filterData(players, GOALKEEPERS)
+                val midfielders = filterData(players, MIDFIELDERS)
+                view.run {
+                    showGoalkeeper(goalkeepers)
+                    showDefender(defenders)
+                    showMidfielder(midfielders)
+                    showForwarder(forwarders)
+                }
+            }
+
+            override fun onFailure(exception: Exception?) {
+                exception?.let {
+                    view.showError(it)
+                }
+            }
+        })
+    }
+
+    private fun filterData(list: List<Player>, type: String) =
+        list.filter { item -> item.playerType == type }
```

Comment:
Đoạn này t nghĩ là ông nên viết theo OOP thì sẽ hay hơn 
bên ngoài không cần rõ type của nó là gì 
```suggestion
        list.filter { item -> item.item.isDefenders() }
```